mcs/class/Microsoft.Build.Engine/Test/Microsoft.Build.BuildEngine/ProjectTest.cs

```diff
@@ -2231,6 +2259,7 @@ public void TestBuildSolutionProject ()
 			}
 		}
 
+#if net_4_0
```

Comment:
This needs to be all uppercase (`NET_4_0`), the #if constants are case-sensitive.